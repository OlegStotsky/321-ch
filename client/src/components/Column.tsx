import * as React from "react";
import ColumnNode from "./ColumnNode";

interface IColumnProps {
  headingMessage: string;
  items: React.ReactNode[];
}

const Column: React.SFC<IColumnProps> = ({ headingMessage, items }) => {
  return (
    <ul className="column">
      <li className="column__heading">{headingMessage}</li>
      {items.map((item, i) => <ColumnNode key={i}>{item}</ColumnNode>)}
    </ul>
  );
};

export default Column;
