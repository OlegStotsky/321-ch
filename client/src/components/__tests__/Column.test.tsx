import * as React from "react";
import * as Enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import Column from "../Column";

Enzyme.configure({ adapter: new Adapter() });
describe("Column", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = Enzyme.shallow(<Column headingMessage="abcd" items={["abcd"]} />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
