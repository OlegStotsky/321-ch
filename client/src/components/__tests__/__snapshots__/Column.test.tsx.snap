// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    headingMessage="abcd"
    items={
      Array [
        "abcd",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="column__heading"
        >
          abcd
        </li>,
        Array [
          <ColumnNode>
            abcd
          </ColumnNode>,
        ],
      ],
      "className": "column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "abcd",
          "className": "column__heading",
        },
        "ref": null,
        "rendered": "abcd",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": "abcd",
        },
        "ref": null,
        "rendered": "abcd",
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="column__heading"
          >
            abcd
          </li>,
          Array [
            <ColumnNode>
              abcd
            </ColumnNode>,
          ],
        ],
        "className": "column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "abcd",
            "className": "column__heading",
          },
          "ref": null,
          "rendered": "abcd",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": "abcd",
          },
          "ref": null,
          "rendered": "abcd",
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
